credentials/apps/edx_credentials_extensions/edly_credentials_app/api/v1/views/edly_sites.py

```diff
@@ -0,0 +1,81 @@
+"""
+Views for Edly Site Creation API.
+"""
+from django.contrib.sites.models import Site
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from credentials.apps.core.models import SiteConfiguration
+from credentials.apps.edx_credentials_extensions.edly_credentials_app.api.permissions import CanAccessSiteCreation
+from credentials.apps.edx_credentials_extensions.edly_credentials_app.api.v1.constants import ERROR_MESSAGES
+from credentials.apps.edx_credentials_extensions.edly_credentials_app.helpers import (
+    get_credentials_site_configuration,
+    validate_site_configurations
+)
+
+
+class EdlySiteViewSet(APIView):
+    """
+    Creates credentials site and it's site configuration.
+    """
+    permission_classes = [IsAuthenticated, CanAccessSiteCreation]
+
+    def post(self, request):
+        """
+        POST /api/edly_api/v1/edly_sites.
+        """
+        validations_messages = validate_site_configurations(request.data)
+        if len(validations_messages) > 0:
+            return Response(validations_messages, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            self.process_client_sites_setup()
+            return Response(
+                {'success': ERROR_MESSAGES.get('CLIENT_SITES_SETUP_SUCCESS')},
+                status=status.HTTP_200_OK
+            )
+        except TypeError:
+            return Response(
+                {'error': ERROR_MESSAGES.get('CLIENT_SITES_SETUP_FAILURE')},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+    def process_client_sites_setup(self):
+        """
+        Process client sites setup and update configurations.
+        """
+        edly_slug = self.request.data.get('edly_slug', '')
+        credentials_base = self.request.data.get('credentials_site', '')
+        theme_dir_name = self.request.data.get('theme_dir_name', 'openedx')
+        lms_url_root = '{protocol}://{lms_url_root}'.format(
+            protocol=self.request.data.get('protocol', 'https'),
+            lms_url_root=self.request.data.get('lms_site', '')
+        )
+        catalog_api_url = '{protocol}://{catalog_api_url}'.format(
```

Comment:
Instead of getting the complete url, we must get the `catalog_site` from the request. In future we don't have to introduce another field if we only need site url. 

```
catalog_api_url = '{protocol}://{catalog_site}/api/v1/'.format(
            protocol=self.request.data.get('protocol', 'https'),
            catalog_site=self.request.data.get('catalog_site', '')
        )

Reply:
True, updating.

Reply:
@burhandodhy Updated.